extensions/mssql/src/contextProvider.ts

```diff
@@ -17,7 +17,8 @@ export enum ContextKeys {
 	ISCLOUD = 'mssql:iscloud',
 	EDITIONID = 'mssql:engineedition',
 	ISCLUSTER = 'mssql:iscluster',
-	SERVERMAJORVERSION = 'mssql:servermajorversion'
+	SERVERMAJORVERSION = 'mssql:servermajorversion',
+	ISSQLONDEMAND = 'mssql:issqlondemand',
```

Comment:
Why are you adding a new var here when you could just have code checking the engine edition? Just seems like unnecessary duplication.

Reply:
I was following the isCloud pattern. If I understand the suggestion correctly - it seems cleaner to check issqlondemand instead of having engine edition comparisons all over the code

Reply:
isCloud is special because there isn't a single edition that means it's a cloud DB. With OnDemand you don't have that concern so we shouldn't need to do that - it's unnecessary and is just adding unnecessary duplication

---
In reply to: [335547853](https://github.com/microsoft/azuredatastudio/pull/7695#discussion_r335547853) [](ancestors = 335547853)